lib/jobs/dell-racadm-set-bios-job.js

```diff
@@ -0,0 +1,111 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = racadmSetBIOSJobFactory;
+di.annotate(racadmSetBIOSJobFactory, new di.Provide('Job.DellRacadm.SetBIOS'));
+di.annotate(racadmSetBIOSJobFactory, new di.Inject(
+    'Job.Base',
+    'JobUtils.RacadmTool',
+    'Logger',
+    'Util',
+    'Assert',
+    'Promise',
+    '_',
+    'Services.Encryption',
+    'Services.Lookup',
+    'Constants',
+    'Services.Waterline'
+));
+
+function racadmSetBIOSJobFactory(
+    BaseJob,
+    racadmTool,
+    Logger,
+    util,
+    assert,
+    Promise,
+    _,
+    encryption,
+    lookup,
+    Constants,
+    waterline
+) {
+    var logger = Logger.initialize(racadmSetBIOSJobFactory);
+    /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function RacadmSetBiosJob(options, context, taskId) {
+        RacadmSetBiosJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = this.context.target;
+        this.userName= options.username;
+        this.passWord = options.password;
+        this.filePath = options.filePath;
+        this.cifsInfo = {user: this.userName, password: this.passWord , filePath: this.filePath};
+
+    }
+    util.inherits(RacadmSetBiosJob, BaseJob);
+
+    /**
+     * @memberOf RacadmSetBiosJob
+     */
+    RacadmSetBiosJob.prototype._run = function() {
+        var self = this;
+        waterline.nodes.findByIdentifier(self.nodeId)
+            .then(function (node) {
+                assert.ok(node, 'No node for dell racadm set bios');
+                var obmSetting = _.find(node.obmSettings, { service: 'ipmi-obm-service' });
+                assert.ok(obmSetting, 'No ipmi obmSetting for dell racadm set bios');
+                return obmSetting.config;
+            })
+            .then(self.revealSecrets)
+            .then(self.lookupHost)
+            .then(function(obmSetting) {
+                return racadmTool.setBiosConfig(obmSetting.host, obmSetting.user,
+                    obmSetting.password,
+                    self.cifsInfo);
+            })
+            .then(function(results){
+                assert.equal(results.jobStatus.status,"Completed",results.jobStatus.status);
+                self._done();
+            })
+            .catch(function(err) {
+                logger.error('The Error Is Found Here!', {error: err});
```

Comment:
This error message "The Error Is Found Here" is not good, it tells nothing to debug